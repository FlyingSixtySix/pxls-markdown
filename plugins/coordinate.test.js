const { processor, expectProcessorParse } = require('../testUtil.js');
const coordinate = require('./coordinate.js');

processor.use(coordinate);

test('does not tokenize "(no coordinates)"', () => expectProcessorParse('(no coordinates)').toMatchSnapshot());
test('tokenizes "(10, 20)" correctly', () => expectProcessorParse('(10, 20)').toMatchSnapshot());
test('tokenizes "(10, 20, 30)" correctly', () => expectProcessorParse('(10, 20, 30)').toMatchSnapshot());
