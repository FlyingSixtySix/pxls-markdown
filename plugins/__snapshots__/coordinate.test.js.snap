// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not tokenize "(no coordinates)" 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "text",
        "value": "(no coordinates)",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "(10, 20)" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "type": "text",
            "value": "(10, 20)",
          },
        ],
        "scale": null,
        "type": "coordinate",
        "url": "#x=10&y=20",
        "x": 10,
        "y": 20,
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "(10, 20, 30)" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "type": "text",
            "value": "(10, 20, 30)",
          },
        ],
        "scale": 30,
        "type": "coordinate",
        "url": "#x=10&y=20&scale=30",
        "x": 10,
        "y": 20,
      },
    ],
    "type": "paragraph",
  },
]
`;
